plugins/modem-manager/fu-mm-backend.c

```diff
@@ -200,36 +200,26 @@ fu_mm_backend_probe_gtype_fallback(FuMmBackend *self, MMObject *omodem, GError *
 		return NULL;
 	}
 
-	if (!mm_modem_get_ports(modem, &used_ports, &n_used_ports)) {
-		g_set_error_literal(error,
-				    FWUPD_ERROR,
-				    FWUPD_ERROR_NOT_SUPPORTED,
-				    "failed to get port information");
-		return NULL;
-	}
-	for (guint i = 0; i < n_used_ports; i++) {
-		g_debug("found port %s: %s",
-			used_ports[i].name,
-			fu_mm_device_port_type_to_string(used_ports[i].type));
-		FU_BIT_SET(ports_bitmask, used_ports[i].type);
+	if (mm_modem_get_ports(modem, &used_ports, &n_used_ports)) {
+		for (guint i = 0; i < n_used_ports; i++) {
```

Comment:
can we move the used_ports and n_used_ports inside the `if {` please.

Reply:
We need to pass the pointers to the function call within the `if ()` or which `if` exactly do you mean?

Reply:
Ohh ignore me, sorry.